### PR TITLE
[backend/frontend] Ability to see 100 simulations in the overview of a scenario in 0.5 seconds

### LINKED ISSUE #1911: Ability to see 100 simulations in the overview of a scenario in 0.5 second
## Description

In the overview of my scenario, I can see my list of 100 simulations loaded in 0.5 second.

### PR DESCRIPTION
**Pull request by isselparra**
### Proposed changes

* Display 10 latest simulations results in Scenario Overview Chart
* Display asynchronously the global score of the list of simulations (scenario overview)

### Related issues

* Closes #1911 

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

**Loading times**
Data: 100 simulations with 50 injects each

Branch: issue/1911
 
http://localhost:3001/api/scenarios/2de41219-c7c6-48b3-8edf-e41781efd105/exercises/search
34.11 ms (8.5 times faster)
 
http://localhost:3001/api/scenarios/2de41219-c7c6-48b3-8edf-e41781efd105/statistics (new)
76.81ms (4.4 times faster)
 
Branch: release/1.10.0
 
http://localhost:3001/api/scenarios/2de41219-c7c6-48b3-8edf-e41781efd105/exercises/search
291.62 ms
 
http://localhost:3001/api/scenarios/2de41219-c7c6-48b3-8edf-e41781efd105/exercises
338.27 ms

### CODE REVIEW

File: openbas-api/src/main/java/io/openbas/rest/exercise/service/ExerciseService.java [hunk 1]
```diff
@@ -431,69 +466,127 @@ private List<ExerciseSimple> execution(TypedQuery<Tuple> query) {
...
-      }
+  private void setComputedAttributes(List<ExerciseSimple> originalExercises) {
+    List<ExerciseSimple> exercises = getExercisesWithId(originalExercises);
+    if (exercises.isEmpty()) {
+      return;
+    }
+
+    Set<String> exerciseIds = getExerciseIds(exercises);
+    MappingsByExerciseIds mappingsByExerciseIds =
+        getResultsByExerciseIds(getExerciseIds(exercises));
```
Comment (savacano28) [inline openbas-api/src/main/java/io/openbas/rest/exercise/service/ExerciseService.java:477]: getExercises()-> replace for 475l

File: openbas-api/src/main/java/io/openbas/rest/exercise/service/ExerciseService.java [hunk 2]
```diff
@@ -431,69 +466,127 @@ private List<ExerciseSimple> execution(TypedQuery<Tuple> query) {
...
+  private static void setGlobalScore(
+      ExerciseSimple exercise, Map<String, List<RawInjectExpectation>> expectationsByExerciseIds) {
+    exercise.setExpectationResultByTypes(
+        AtomicTestingUtils.getExpectationResultByTypesFromRaw(
+            expectationsByExerciseIds.getOrDefault(exercise.getId(), emptyList())));
+  }
+
+  private static void setEmptyGlobalScore(ExerciseSimple exercise) {
+    exercise.setExpectationResultByTypes(new ArrayList<>());
+  }
```
Comment (savacano28) [inline openbas-api/src/main/java/io/openbas/rest/exercise/service/ExerciseService.java:563]: Maybe for this one line, we don't need to extract it into a function.

File: openbas-api/src/main/java/io/openbas/rest/exercise/service/ExerciseService.java (hunk 1 above)
Comment (isselparra) [reply openbas-api/src/main/java/io/openbas/rest/exercise/service/ExerciseService.java]: Done

File: openbas-api/src/main/java/io/openbas/rest/exercise/service/ExerciseService.java (hunk 2 above)
Comment (isselparra) [reply openbas-api/src/main/java/io/openbas/rest/exercise/service/ExerciseService.java]: Done

File: openbas-api/src/test/java/io/openbas/rest/scenario/service/ScenarioStatisticServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
```
Comment (savacano28) [inline openbas-api/src/test/java/io/openbas/rest/scenario/service/ScenarioStatisticServiceTest.java:28]: Remove

File: openbas-api/src/test/java/io/openbas/utils/ResultUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import java.util.Set;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
```
Comment (savacano28) [inline openbas-api/src/test/java/io/openbas/utils/ResultUtilsTest.java:23]: Remove

File: openbas-api/src/test/java/io/openbas/rest/scenario/service/ScenarioStatisticServiceTest.java (hunk 3 above)
Comment (isselparra) [reply openbas-api/src/test/java/io/openbas/rest/scenario/service/ScenarioStatisticServiceTest.java]: Done

Comment (isselparra) [reply openbas-api/src/test/java/io/openbas/rest/scenario/service/ScenarioStatisticServiceTest.java]: Done

File: openbas-api/src/test/java/io/openbas/utils/ResultUtilsTest.java (hunk 4 above)
Comment (isselparra) [reply openbas-api/src/test/java/io/openbas/utils/ResultUtilsTest.java]: Done

File: openbas-front/src/admin/components/scenarios/scenario/Scenario.tsx [hunk 5]
```diff
@@ -63,26 +60,19 @@ const Scenario = ({ setOpenInstantiateSimulationAndStart }: { setOpenInstantiate
...
     scenario: helper.getScenario(scenarioId),
-    exercises: helper.getExercisesMap(),
   }));
-  const [loadingScenarioExercises, setLoadingScenarioExercises] = useState(true);
-  useDataLoader(() => {
-    setLoadingScenarioExercises(true);
-    dispatch(fetchScenarioExercises(scenarioId)).finally(() => setLoadingScenarioExercises(false));
-  });
-  const scenarioExercises = scenario.scenario_exercises?.map((exerciseId: string) => exercisesFromStore[exerciseId]).filter((ex: ExerciseStore) => !!ex);
+  const scenarioHasExercises = scenario.scenario_exercises?.length > 0;
```
Comment (savacano28) [inline openbas-front/src/admin/components/scenarios/scenario/Scenario.tsx:68]: maybe hasScenarioExercises?

**Review by savacano28 — APPROVED** [review]
Tested ok!
Great job!

File: openbas-front/src/admin/components/scenarios/scenario/Scenario.tsx (hunk 5 above)
Comment (isselparra) [reply openbas-front/src/admin/components/scenarios/scenario/Scenario.tsx]: Done

**Pull request merged**